pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -0,0 +1,563 @@
+			})
+			It("should create deployment and set owner", func() {
+				deployment := appsv1.Deployment{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, &deployment)
+				Expect(err).To(HaveOccurred())
+
+				ReconcileFully(client, *controller, false)
+
+				err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, &deployment)
+				Expect(err).ToNot(HaveOccurred())
+
+				testSpecs := NewFakeServiceSpecs(false)
+				testTLSConfig := NewFakeTLSConfig()
+				expectedDeployment := resource_definitions.NewDeploymentForCR(test.NewContainerJFR(), &testSpecs, &testTLSConfig)
+				Expect(deployment.ObjectMeta.Name).To(Equal(expectedDeployment.ObjectMeta.Name))
+				Expect(deployment.ObjectMeta.Namespace).To(Equal(expectedDeployment.ObjectMeta.Namespace))
+				Expect(deployment.ObjectMeta.Labels).To(Equal(expectedDeployment.ObjectMeta.Labels))
+				Expect(deployment.ObjectMeta.Annotations).To(Equal(expectedDeployment.ObjectMeta.Annotations))
+				Expect(deployment.Spec.Selector).To(Equal(expectedDeployment.Spec.Selector))
+
+				// compare Pod template
+				template := deployment.Spec.Template
+				expectedTemplate := expectedDeployment.Spec.Template
+				Expect(template.ObjectMeta).To(Equal(expectedTemplate.ObjectMeta))
+				Expect(template.Spec.Containers).To(Equal(expectedTemplate.Spec.Containers))
+				Expect(template.Spec.Volumes).To(Equal(expectedTemplate.Spec.Volumes))
+			})
+		})
+		Context("succesfully creates required resources for minimal deployment", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewMinimalContainerJFR(),
+				}
+			})
+			It("should create persistent volume claim and set owner", func() {
+				pvc := &corev1.PersistentVolumeClaim{}
+				err := client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, pvc)
+				Expect(err).To(HaveOccurred())
+
+				ReconcileFully(client, *controller, true)
+
+				err = client.Get(context.Background(), types.NamespacedName{Name: "containerjfr", Namespace: "default"}, pvc)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Compare to desired spec
+				expectedPvc := resource_definitions.NewPersistentVolumeClaimForCR(test.NewContainerJFR())
+				Expect(pvc.ObjectMeta.Name).To(Equal(expectedPvc.ObjectMeta.Name))
+				Expect(pvc.ObjectMeta.Namespace).To(Equal(expectedPvc.ObjectMeta.Namespace))
+				Expect(pvc.Spec.AccessModes).To(Equal(expectedPvc.Spec.AccessModes))
+				Expect(pvc.Spec.StorageClassName).To(Equal(expectedPvc.Spec.StorageClassName))
+
+				pvcStorage := pvc.Spec.Resources.Requests["storage"]
+				expectedPvcStorage := expectedPvc.Spec.Resources.Requests["storage"]
+				Expect(pvcStorage.Equal(expectedPvcStorage)).To(BeTrue())
+
+				// Check for owner
+				Expect(pvc.ObjectMeta.OwnerReferences[0].Kind).To(Equal("ContainerJFR"))
+				Expect(pvc.ObjectMeta.OwnerReferences[0].Name).To(Equal("containerjfr"))
```

Comment:
If possible, could you refactor the specs (the body of the `It` calls) to helpers to reduce duplication between the minimal and non-minimal Contexts?